vision/src/autogluon/vision/configs/presets_configs.py

```diff
@@ -0,0 +1,184 @@
+"""Presets for vision predictors"""
+import functools
+from autogluon.core import Categorical, Real
+
+
+def unpack(preset_name):
+    if not preset_name in _PRESET_DICTS:
+        raise ValueError(f'Unknown preset_name: {preset_name}')
+    def _unpack_inner(f):
+        @functools.wraps(f)
+        def _call(*args, **kwargs):
+            gargs, gkwargs = set_presets(preset_name, *args, **kwargs)
+            return f(*gargs, **gkwargs)
+        return _call
+    return _unpack_inner
+
+
+# Dictionary of preset fit() parameter configurations for ImagePredictor.
+preset_image_predictor = dict(
+    # Best predictive accuracy with little consideration to inference time or model size. Achieve even better results by specifying a large time_limit value.
+    # Recommended for applications that benefit from the best possible model accuracy.
+    best_quality={
```

Comment:
This will end very poorly if someone does not specify a time_limit. We should ensure that a time limit is set for this, otherwise it will run 1024 trials.

Reply:
Same comments for all of the presets. Any time num_trials is specified automatically, we probably want to ensure time_limit is specified by user

Reply:
here's my confusing part, if running things without a wall clock time limit is scary, should we at least safeguard the default time_limit with a reasonable value, say 1 day or 3 days?

Reply:
Perhaps we can set default to 4 hours in the presets and warn the user if they did not set it themselves. Whatever solution we go with, we need to ensure that the user doesn't unknowingly start a job by specifying a preset that will take weeks+ to finish when they never touched time_limit.

I would personally be in favor of raising an exception if the user specified a preset but did not specify time_limit, or otherwise set num_trials=1 if they only specified a preset but no time limit. Alternatively, we could set time_limit default value in the preset itself (and user can freely override if they wish).

Example:

```
    best_quality={
        'hyperparameters': {
            'model': Categorical('resnet50_v1b', 'resnet101_v1d', 'resnest200'),
            'lr': Real(1e-2, 1e-6, log=True),
            'batch_size': Categorical(8, 16, 32, 64, 128),
            'epochs': 200
            },
        'hyperparameter_tune_kwargs': {
            'num_trials': 1024,
            'search_strategy': 'bayesopt',
            'max_reward': 1.0
        },
        'time_limit': 86400,
    },
```

Reply:
Alternative suggestion:

- Do not set time-limits by default (since users expect things to take longer for larger datasets anyway).

- Set num_trials default (probably much lower than 1024 though. Perhaps for the different presets we could have: num_trials = 1 (for medium), 10 (for good), 100 (for best).

- Whenever user specifies time_limit, this overrides num_trials. So we don't run for min{time_limit, num_trials}, but instead just run for whatever time_limit the user specified (which may involve more trials than the default num_trials). If user specifies both time_limit and num_trials, then throw Exception.